cpp/turbodbc/Library/src/make_description.cpp

```diff
@@ -101,10 +102,11 @@ template <typename Description>
 std::unique_ptr<description> make_character_description(cpp_odbc::column_description const & source,
                                                         turbodbc::options const & options)
 {
+    int multiplier = (options.fix_unicode_allocation ? Description::multiplier : 1);
     std::size_t const sanitized_size = (source.size == 0 ? options.varchar_max_character_limit : source.size);
     std::size_t const limited_size = (options.limit_varchar_results_to_max ?
                                       std::min(sanitized_size, options.varchar_max_character_limit) :
-                                      sanitized_size);
+                                      sanitized_size) * multiplier;
```

Comment:
Interesting question whether the additional capacity should be added before or after sanitation. Your choice is alright, but the interaction between `varchar_max_character_limit`  and `force_extra_capacity_for_unicode` should be documented somewhere, perhaps in the `advanced_usage.rst` in the `docs/pages` folder.

Reply:
If the capacity modifier is applied before the restriction, we are no longer restricting characters but code points, which would be unexpected and lead to unexpected results, no?

Reply:
Likewise, it could be unexpected that turbodbc requires more memory (in bytes) than the specified number of "characters" (whatever that means :-)).

But as I said, your choice is alright, and capped strings are probably more confusing than higher memory requirements. I still feel this needs externally visible documentation, though.

Reply:
Yep, writing it up now!